packages/react/src/primitives/ToggleButtonGroup/useToggleButtonGroup.tsx

```diff
@@ -42,19 +44,19 @@ export const useToggleButtonGroup = ({
   );
 
   // Exclusive selection
-  const handleExclusiveChange: ToggleButtonProps['onChange'] =
-    React.useCallback(
-      (buttonValue) => {
-        if (!isFunction(onChange)) {
-          return;
-        }
-
-        onChange(
-          value === buttonValue && !isSelectionRequired ? null : buttonValue
-        );
-      },
-      [onChange, value, isSelectionRequired]
-    );
+  const handleExclusiveChange = React.useCallback<
+    (buttonValue: string | undefined) => void
+  >(
+    (buttonValue) => {
+      if (!isFunction(onChange) || Array.isArray(value)) {
```

Comment:
Could be wrong but adding the `Array.isArray(value)` condition reads like different behavior from what we currectly have?